core/src/wallet/cosmos/api_impl/CosmosLikeReward.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ * CosmosLikeReward
+ *
+ * Created by Hakim Aammar on 17/03/2020.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <wallet/cosmos/api_impl/CosmosLikeReward.hpp>
+#include <wallet/cosmos/CosmosLikeCurrencies.hpp>
+#include <wallet/common/Amount.h>
+
+#include <cmath>
+
+
+
+namespace ledger {
+    namespace core {
+
+        CosmosLikeReward::CosmosLikeReward(const cosmos::Reward& rewardData, const std::string& delegatorAddress) :
+            _rewardData(rewardData),
+            _delegatorAddress(delegatorAddress)
+        {}
+
+        std::string CosmosLikeReward::getDelegatorAddress() const {
+            return _delegatorAddress;
+        }
+
+        std::string CosmosLikeReward::getValidatorAddress() const {
+            return _rewardData.validatorAddress;
+        }
+
+        std::shared_ptr<api::Amount> CosmosLikeReward::getRewardAmount() const {
+            // FIXME std::stod uses radix from current locale (can be . or ,)
+            // resulting in incorrect rounding for any locale where the radix is not '.'
+            auto roundedReward = std::to_string(std::lround(std::stod(_rewardData.pendingReward.amount)));
```

Comment:
I had the same issue when fetching the total rewards for an account.
Better not use the locale and just truncate the string using `find('.')`

Reply:
Yeah I thought about this.
- Rounding like this makes the result inconsistent: precise on some environments, and off on others, depending on the locale.
- Truncating would make the result consistent, but never exact, always off by a few units... (can be as much as 1 unit per delegation)
Do you prefer the latter?...

Reply:
it's truncating at most 1 microATOM, users won't care that much I think

Reply:
Fair enough, truncate it is :)